shapefil.h

```diff
@@ -292,13 +292,13 @@ extern "C"
     SHPHandle SHPAPI_CALL SHPCreate(const char *pszShapeFile, int nShapeType);
     SHPHandle SHPAPI_CALL SHPCreateLL(const char *pszShapeFile, int nShapeType,
                                       const SAHooks *psHooks);
-    void SHPAPI_CALL SHPGetInfo(SHPHandle hSHP, int *pnEntities,
+    void SHPAPI_CALL SHPGetInfo(const SHPHandle hSHP, int *pnEntities,
                                 int *pnShapeType, double *padfMinBound,
                                 double *padfMaxBound);
 
-    SHPObject SHPAPI_CALL1(*) SHPReadObject(SHPHandle hSHP, int iShape);
-    int SHPAPI_CALL SHPWriteObject(SHPHandle hSHP, int iShape,
-                                   SHPObject *psObject);
+    SHPObject SHPAPI_CALL1(*) SHPReadObject(const SHPHandle hSHP, int iShape);
+    int SHPAPI_CALL SHPWriteObject(const SHPHandle hSHP, int iShape,
```

Comment:
const on the SHPHandle is odd for a writing function:

```suggestion
    int SHPAPI_CALL SHPWriteObject(SHPHandle hSHP, int iShape,
```